[Backend Tester] Add test flows for QNN

GregoryComer opened the pull request:
Add test flows for each of the supported QNN quantization modes. I followed the guidance in the docs and example code for the overall lowering flow. Add jobs to nightly CI.

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

digantdesai commented on backends/qualcomm/tests/tester.py:
```diff
@@ -73,15 +98,17 @@ def __init__(
-        )
+        stage_classes = executorch.backends.test.harness.Tester.default_stage_classes() | {
+            StageType.PARTITION: Partition,
+            StageType.TO_EDGE_TRANSFORM_AND_LOWER: create_to_edge_transform_and_lower,
```
So not even generating a PTE?

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
[Backend Tester] Add QNN test flow to nightly CI (#13468)

Add (unquantized) QNN test flows to the nightly CI. This runs both the
model and operator suite.

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

cccclai commented on .github/workflows/nightly.yml:
```diff
@@ -42,7 +42,7 @@ jobs:
       fail-fast: false
       matrix:
-        flow: [qualcomm, vulkan, xnnpack, xnnpack_static_int8_per_channel]
+        flow: [qualcomm, qualcomm_16a16w, qualcomm_16a8w, qualcomm_16a4w, qualcomm_16a4w_block, qualcomm_8a8w, vulkan, xnnpack, xnnpack_static_int8_per_channel]
```
Can we rename to qnn_* to align with the naming convention

GregoryComer replied:
Sure, I'll update it.

GregoryComer replied:
> why not also add more for xnnpack? Just curious

Added a few more PT2E flows in the stack. I'll add int4 as well.

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer commented:
Note that CI failures are pre-existing or infra failures.
